### PR TITLE
Enable SSL verification for all requests

### PR DESCRIPTION
**Pull request by dkorn**
By default, SSL verification is enabled in requests HTTP python library, which is used by this client.
This change re-enables the verification so that it would be possible to use against cert configured manageiq environments. To use a self-signed certificate pass the certificate file path to the API class: `ca_bundle_path='/path/to/certfile'`.

Ignoring verifying the SSL certificate is possible by instantiating
the API class with: `verify_ssl=False`.

ca-certificates in python-requests [docs](http://docs.python-requests.org/en/master/user/advanced/#ca-certificates)

### CODE REVIEW

**Comment by dkorn** [comment]
@cben @simon3z @enoodle please review
cc @twiest @kwoodson

File: src/miqclient/api.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class APIException(Exception):
 
 
 class API(object):
-    def __init__(self, entry_point, auth, logger=None):
+    def __init__(self, entry_point, auth, logger=None, verify=True, certfile_path=None):
```
Comment (cben) [inline src/miqclient/api.py]: Consider s/verify/verify_tls/.
And maybe s/certfile_path/ca_bundle_path/?  Not sure if that's the best name, but a particular drawback of calling it "certfile" is that `requests` supports both file and directory bundles.

It's worth documenting that if the bundle path is specified it _replaces_ the system trust roots.  If you just want to add some cert, you have to concat it to the system's CA bundle.

Comment (dkorn) [reply src/miqclient/api.py]: @cben thanks for the review.
I've made the changes you proposed, with a small difference of s/verify/verify_ssl

**Review by cben — APPROVED** [review]

**Comment by simon3z** [comment]
:+1: (I haven't reviewed the code properly though)

**Comment by simon3z** [comment]
@dkorn if `verify_ssl` is `True` but `ca_bundle_path` is `None` what is happening? Should you explicitly validate that `ca_bundle_path` is not `None`?

**Comment by cben** [comment]
That's OK, in fact that's the default — it uses the system's CA bundle.
`ca_bundle_path` is mainly useful if you want to user a self-signed cert.

**Comment by dkorn** [comment]
@simon3z what @cben said :) 
I added a comment about it in the class docstring

**Comment by simon3z** [comment]
@dkorn @cben have you synced with @mfalesni? I assume he'll have to specifically set this to `False` if his automation switched to use this python package already.

**Comment by simon3z** [comment]
I am merging this but please get in contact with @mfalesni ASAP to sort out if he needs changes on his side.

**Pull request merged**